silkworm/rpc/http/connection.cpp

```diff
@@ -428,4 +425,31 @@ std::string Connection::get_date_time() {
     return ss.str();
 }
 
+Task<bool> Connection::compress(
+    boost::asio::thread_pool& workers,
+    const std::string& clear_data,
+    std::string& compressed_data) {
+    auto this_executor = co_await boost::asio::this_coro::executor;
+    boost::iostreams::filtering_ostream out;
+    const auto ret_success = co_await boost::asio::async_compose<decltype(boost::asio::use_awaitable), void(bool)>(
+        [&](auto& self) {
+            boost::asio::post(workers, [&, self = std::move(self)]() mutable {
+                bool success = true;
+                try {
+                    out.push(boost::iostreams::gzip_compressor());
+                    out.push(boost::iostreams::back_inserter(compressed_data));
+                    boost::iostreams::copy(boost::make_iterator_range(clear_data), out);
+                } catch (const std::exception& e) {
```

Comment:
This should catch also `boost::system::system_error` and extract error code to return back to caller using `self.complete`